stack-overflow/src/model/VoteQuestion.js

```diff
@@ -0,0 +1,105 @@
+import { EventEmitter } from "events";
+import modelQuestion from "./Question";
+import modelLogin from "./Login";
+
+class VoteQuestion extends EventEmitter {
+    constructor() {
+        super();
+        this.state = {
+            votedQuestions: [{
+
+               
+            }],
+        };
+    }
+
+    addVotedQuestion(userGivesVote, userGetsVote, questionVoted, voteType) {
+        this.state = {
+            ...this.state,
+            votedQuestions: Object.keys(this.state.votedQuestions).concat([{
+                userGivesVote: userGivesVote,
+                userGetsVote: userGetsVote,
+                questionVoted: questionVoted,
+                voteType: voteType
+            }])
+        };
+        this.emit("change", this.state);
+    }
+
+    changeQuestionVoteProperty(property, value, index) {
+        this.state = {
+            ...this.state,
+            votedQuestions: {
+                ...this.state.votedQuestions,
+                [index]: {
+                    ...this.state.votedQuestions[index],
+                    [property]: value
+                }
+            }
+        };
+        this.emit("change", this.state);
+    }
+
+    findVoteForQuestion(userGivesVote, questionToBeVoted, voteText) {
+
+        var isQuestionVoted = false;
+        var differentQuestionVote = false;
+        var index;
+        for (let i = 0; i < Object.keys(this.state.votedQuestions).length; i++) {
+
+            var votedQuestion = this.state.votedQuestions[i];
+
+            if (votedQuestion.userGivesVote === userGivesVote &&
+                votedQuestion.questionVoted.author === questionToBeVoted.author && votedQuestion.questionVoted.text === questionToBeVoted.text &&
+                votedQuestion.questionVoted.date === questionToBeVoted.date && votedQuestion.questionVoted.tags === questionToBeVoted.tags &&
+                votedQuestion.questionVoted.title === questionToBeVoted.title) {
+                if (votedQuestion.voteType !== voteText) {
+
+                    differentQuestionVote = true;
+                    index = i;
+                }
+
+                isQuestionVoted = true;
+                break;
+            }
+        }
+
+        modelQuestion.changeQuestionProperty("allowVote", differentQuestionVote || !isQuestionVoted);
+        modelQuestion.changeQuestionProperty("isQuestionVoted", isQuestionVoted);
+        if (!isQuestionVoted)
```

Comment:
I would recommend always using curly brackets when doing if-else clauses.

Also for readability, I usually put the else and the next if on the same line:

```

if (bla) {
...
} else if (bla) {
...
}

```